HopsanGenerator/src/generators/HopsanGeneratorBase.cpp

```diff
@@ -1056,7 +1056,7 @@ bool HopsanGeneratorBase::copyDefaultComponentCodeToDir(const QString &path) con
     return true;
 }
 
-bool HopsanGeneratorBase::copyExternalComponentCodeToDir(const QString &destinationPath, const QStringList &externalLibraries) const
+bool HopsanGeneratorBase::copyExternalComponentCodeToDir(const QString &destinationPath, const QStringList &externalLibraries, QStringList &extraSourceFiles) const
```

Comment:
is unclear that the extraSourceFiles are meant to be an output variable here, (no documentation block) and I would prefer if it is called rExtraSourceFiles,  (r for reference and return value). (Some code styles would recommend sending by pointer her, but lets not do that since it is not done in other places in Hopsan.)

Reply:
I added a comment and renamed the argument. Sholdn't all other reference arguments in HopsanGenerator also use the "r" prefix then?

Reply:
Yes initially the style was  'type &rParameter' for all references, but I think that we should instead only use that style for references that are also output variables. Then the r can mean both 'return' and 'reference'. Inputs as references should be. 'const type &parameter'. An alternative would be to follow the google styleguide recommendation and allways use pointers for output parameters.

Reply:
Ok, sounds reasonable. We currently use pointers in some places and references in others. But this discussion does not belong in this PR, so I will keep these references for now.